app/scaffold-controllers/welcome-screen/welcomeModalController.js

```diff
@@ -0,0 +1,87 @@
+define([
+    'astro-full',
+    'astro-rpc',
+    'bluebird',
+    'plugins/modalViewPlugin',
+    'plugins/secureStorePlugin',
+    'scaffold-controllers/welcome-screen/welcomeController'
+],
+/* eslint-disable */
+function(
+    Astro,
+    AstroRpc,
+    Promise,
+    ModalViewPlugin,
+    SecureStorePlugin,
+    WelcomeController
+) {
+/* eslint-enable */
+
+    var WelcomeModalController = function(modalView) {
+        this.isShowing = false;
+        this.viewPlugin = modalView;
+    };
+
+    WelcomeModalController.init = function() {
+        return Promise.join(
+            ModalViewPlugin.init(),
+            WelcomeController.init(),
+        function(modalView, welcomeController) {
+            modalView.setContentView(welcomeController.viewPlugin);
+
+            // This registers a close handler on the header bar to dismiss
+            // the modal. Without a header bar, the developer is responsible
+            // for implementing a way to dismiss the modal.
+            var welcomeModalController = new WelcomeModalController(modalView);
+            welcomeController.registerCloseEventHandler(function() {
+                welcomeModalController.hide();
+            });
+
+            // Welcome modal RPCs
+            Astro.registerRpcMethod(AstroRpc.names.showWelcome, [], function(res) {
+                welcomeModalController.show({forced: true});
+            });
+
+            Astro.registerRpcMethod(AstroRpc.names.hideWelcome, [], function(res) {
+                welcomeModalController.hide();
+            });
+
+            return welcomeModalController;
+        });
+    };
+
+    WelcomeModalController.prototype.show = function(params) {
+        var self = this;
+        params = Astro.Utils.extend({forced: false}, params);
+
+        var secureStorePromise = SecureStorePlugin.init();
```

Comment:
Yeah, wasn't thinking we pass it in, but we could init it once for this controller and eliminate some of the promise management here because we'd have a fully resolved secure store plugin by this point.